### DISCUSSION TITLE
Channel.getChatListFrom() 이용시 최신 메시지를 받아 올수 없는 문제

### DISCUSSION THREAD

**Discussion by async3619** [storycraft/node-kakao#695] [Q&A] [answered]
`Channel` 객체의 `getChatListFrom` 메소드를 호출시 메시지 리스트는 잘 업어 옵니다만 최신 메세지들을 누락시킨채로 과거의 메시지만 가져오는것 같습니다. 혹시 가장 최근 메시지까지 다 받아올수 있는 방법은 없는걸까요?

**Comment by generalxd** [comment]
다 잘 받아옵니다.

**Reply by generalxd** [reply]
폭탄 메시지에 대한 정보를 못받아오는거 때문에 질문하신거같은데 syncChatList도 같은 현상이실겁니다.

다른방법을 사용하셔야합니다.

**Reply by async3619** [reply]
@generalxd 아뇨 폭탄 메시지도 아니고 그냥 일반 평범한 계정의 대화방 로그를 가져오는데 이러한 문제가 생기고 있던겁니다

**Comment by storycraft** [comment, marked as answer]
모든 메세지를 전부 받아오는 작업에 `getChatListFrom` 메서드는 적합하지 않습니다.  대신 `syncChatList` 메서드를 사용하세요. 해당 메서드는 지정된 범위의 채팅들을 최대 300개씩 받아올수 있는 비동기 이터레이터를 반환합니다. 가장 최근의 Chatlog또는 logId값은 `ChannelInfo`객체의 `lastChatlogId` 또는 `lastChatlog`값을 통해 알 수 있습니다.

**Reply by async3619** [reply]
제가 글을 제대로 안읽고 재질문을 드렸네요 감사합니다!! 😀😀🙇
